feat: 3.2.0 activation (#466)
bgd-labs/aave-proposals-v3: src/20240924_Multi_UpgradeAllAaveInstancesTo32/UpgradeAllAaveInstancesTo32.md

Stable debt has been off-boarded from Aave v3 for quite some time. What was remaining on the protocol code were merely artifacts that increased gas consumption and code size.
Therefore the 3.2 release removes these artifacts in a backwards compatible way.

## Specification

Aave 3.2 focuses on two main areas of the aave protocol:

- The final deprecation of stable borrowing
- Improvements on the eModes, introducing "Liquid eModes"

_Please note: For existing users, the upgrade is 100% backwards compatible and no migration or similar is required. With introduction of liquid emode, entering and leaving an eMode still works via setUserEMode(categoryId) and getUserEMode(address user) like in previous versions of the protocol._

The proposal will execute to following operations on each active Aave v3 pool:

- Upgrade the Pool/L2Pool implementation to 3.2
- Upgrade the PoolConfigurator implementation to 3.2
- Deploy new ProtocolDataProvider compatible with v3.2, and set it on the addresses provider contract
- Migrate all assets currently in eMode to be both borrowable & collateral in eMode
- Migrate the InterestRateStrategy to a new version without stable rate calculations

> 

Author: put sth about security procedures (who audited -> links to reports)

## References

- [Payload Implementation](https://github.com/bgd-labs/protocol-v3.2-upgrade/blob/main/src/contracts/UpgradePayload.sol)
- [New initialization logic of Pool](https://github.com/bgd-labs/protocol-v3.2-upgrade/blob/main/src/contracts/CustomInitialize.sol)
- [New L1 pool implementation](https://github.com/bgd-labs/protocol-v3.2-upgrade/blob/main/src/contracts/PoolInstance.sol)
- [New L2 pool implementation](https://github.com/bgd-labs/protocol-v3.2-upgrade/blob/main/src/contracts/L2PoolInstance.sol)
- [Payload Tests](https://github.com/bgd-labs/protocol-v3.2-upgrade/tree/main/tests)
- [Code and post execution state diffs](https://github.com/bgd-labs/protocol-v3.2-upgrade/tree/main/diffs)
- [Snapshot](https://snapshot.org/#/aave.eth/proposal/0x68ce69b5e71df1d77c2ad814a5d41162a40be54473576ff590d0b1bb5afde4a7)

> - [Discussion](https://governance.aave.com/t/bgd-aave-v3-2-liquid-emodes/19037/3)

Author: probably makes sense to reference the migration guide

---
title: "Upgrade all Aave instances to 3.2"
author: "BGD Labs @bgdlabs"
discussions: "https://governance.aave.com/t/bgd-aave-v3-2-liquid-emodes/19037/3"
snapshot: "https://snapshot.org/#/aave.eth/proposal/0x68ce69b5e71df1d77c2ad814a5d41162a40be54473576ff590d0b1bb5afde4a7"
---

## Simple Summary

> Upgrades all active Aave instances to Aave 3.2 which debuts "Liquid eModes" and fully deprecated stable borrowing.

Reviewer: ```suggestion
Upgrades all active Aave instances to Aave v3.2, which debuts "Liquid eModes" and fully deprecates stable borrowing.
```

## Motivation

EModes have been a powerful feature since the inception of Aave v3.

> Liquid eModes builds on this success by increase the flexibility in configuring eModes. With Liquid eModes it is now possible, to have a single asset be listed in multiple eModes.
> At the same time, it's possible to granularly control which assets are collateral and borrowable inside an eMode.

Reviewer: ```suggestion
Liquid eModes build on this success by increasing the flexibility when configuring eModes:  now it is possible for a single asset to be listed in multiple eModes.
Additionally, it is possible to granularly control which assets are collateral and/or borrowable inside an eMode.
```

Stable debt has been off-boarded from Aave v3 for quite some time. What was remaining on the protocol code were merely artifacts that increased gas consumption and code size.
Therefore the 3.2 release removes these artifacts in a backwards compatible way.

_In addition to the upgrade itself, the proposal payload includes reimbursement of $76’000 denominated in GHO, to cover the cost of extra external security procedures incurred._

## Specification

> Aave 3.2 focuses on two main areas of the aave protocol:

Reviewer: ```suggestion
Aave 3.2 focuses on two main areas of the Aave Protocol:
```

- The final deprecation of stable borrowing
- Improvements on the eModes, introducing "Liquid eModes"

The proposal will execute to following operations on each active Aave v3 pool:

- Upgrade the Pool/L2Pool implementation to 3.2, which will internally set the stableDebtToken address to `address(0)` on all assets
- Upgrade the PoolConfigurator implementation to 3.2
- Deploy new ProtocolDataProvider compatible with v3.2, and set it on the addresses provider contract
- Migrate all assets currently in eMode to be both borrowable & collateral in eMode
- Migrate the InterestRateStrategy to a new version without stable rate calculations

> An additional payload will transfer `76,000.00` GHO to the BGD-controlled address `0xb812d0944f8F581DfAA3a93Dda0d22EcEf51A9CF`.

Reviewer: ```suggestion
An additional, the payload will transfer `76,000.00` GHO to the BGD-controlled address `0xb812d0944f8F581DfAA3a93Dda0d22EcEf51A9CF`.
```

## Security procedures

> In addition to unit tests & integration test suites, the changes have been audited in 5 audits by 4 independent auditors:

Reviewer: ```suggestion
On top of unit tests & integration test suites, the codebase has undergone 5 security audits by 4 independent third-party experts:
```
